llvm/lib/Transforms/Vectorize/LoopVectorize.cpp

```diff
@@ -5253,7 +5253,7 @@ LoopVectorizationCostModel::calculateRegisterUsage(ArrayRef<ElementCount> VFs) {
       OpenIntervals.erase(ToRemove);
 
     // Ignore instructions that are never used within the loop.
-    if (!Ends.count(I))
+    if (!Ends.count(I) && !I->mayHaveSideEffects())
```

Comment:
I guess this may also include calls to intrinsics or functions that return void?